mapper/dm_agent/src/batcher.rs

```diff
@@ -382,12 +383,19 @@ mod tests {
         ]);
 
         let first_message = message_stream.next().await.unwrap();
-        let builder = MessageBatcher::new(sender, Arc::new(mqtt_client)).unwrap();
+        let time_provider = WallClock;
+        let builder = MessageBatcher::new(
+            sender,
+            Arc::new(mqtt_client),
+            Duration::from_millis(1000),
+            TopicFilter::new("collectd/#").unwrap().qos(QoS::AtMostOnce),
+            Box::new(time_provider.clone()),
```

Comment:
I think an `Arc` would be generally better than cloning the `time_provider`. So just replace every occurence of `Box<dyn Clock>` with `Arc<dyn Clock>`.